sdcm/utils/raft/common.py

```diff
@@ -131,26 +141,114 @@ def run_bootstrap_and_abort_with_action(self, terminate_pattern, abort_action: C
-
-                self.bootstrap_node.start_scylla_server(verify_up_timeout=bootstrap_timeout, verify_down=True)
-                self.bootstrap_node.start_scylla_jmx()
-            self.db_cluster.check_nodes_up_and_normal(
-                nodes=[self.bootstrap_node], verification_node=self.verification_node)
+            LOGGER.debug("Start rebootstrap as new node")
+            ParallelObject(objects=[self._rebootstrap_node, watcher_startup_failed], timeout=3800).call_objects()
+            LOGGER.debug("Node is up")
+        except NodeSetupFailed as exc:
+            LOGGER.error("Scylla service restart failed: %s", exc)
+            self.clean_unbootstrapped_node()
+            raise BootstrapStreamErrorFailure(f"Rebootstrap failed with error: {exc}") from exc
+        except ExitByEventError as exc:
+            LOGGER.error("Event stopped: %s", exc)
+            if self.bootstrap_node.db_up():
+                LOGGER.info("Node is up")
+            else:
+                LOGGER.info("Clean node")
+                self.clean_unbootstrapped_node()
+
         except Exception as exc:  # pylint: disable=broad-except  # noqa: BLE001
             LOGGER.error("Scylla service restart failed: %s", exc)
             self.clean_unbootstrapped_node()
             raise BootstrapStreamErrorFailure(f"Rebootstrap failed with error: {exc}") from exc
+        finally:
+            self.bootstrap_node.stop_wait_db_up_event.clear()
+
+
+class MonitorDecommissionOperation:
+    DECOMMISSION_STATE = "DECOMMISSIONING"
+
+    def __init__(self, target_node: BaseNode, verification_node: BaseNode, timeout=7200):
+        self.timeout = timeout
+        self.target_node = target_node
+        self.db_cluster: BaseScyllaCluster = target_node.parent_cluster
+        self.target_node_ip = target_node.ip_address
+        self.verification_node = verification_node
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if exc_type:
+            LOGGER.warning("Decommission failed with error %s: %s\n%s", exc_type, exc_val, exc_tb)
+            LOGGER.debug("Check is decommission running..")
+            if self.is_node_decommissioning():
+                wait_for(func=lambda: not self.is_node_decommissioning(), step=15,
+                         timeout=self.timeout,
+                         text=f"Waiting decommission is finished for {self.target_node.name}...")
+                self.db_cluster.verify_decommission(self.target_node)
+                return True
+            elif not self.is_node_decommissioning():
+                self.db_cluster.verify_decommission(self.target_node)
+                return True
+
+    def is_node_decommissioning(self):
+        node_status = self.db_cluster.get_node_status_from_system_by(
+            self.verification_node, ip_address=self.target_node_ip)
+        LOGGER.info("New node %s state after decommission failed: %s", self.target_node.name, node_status)
```

Comment:
this log line will be repeated over and over, move it to debug or remove - as anyway other log lines will tell the status of the node.

Reply:
removed it, common cal already have this info